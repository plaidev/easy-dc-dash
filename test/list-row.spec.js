import Vue from 'vue';
import { mount, createLocalVue } from 'vue-test-utils';
import CssModuleTestHelperMixin from './helpers/css-modules-test-helper-mixin';

import Store from '../libs/store';
import ListRow from '@/list-row.vue';

const localVue = createLocalVue();
localVue.use(CssModuleTestHelperMixin);

describe('list-row', () => {
  Store.registerData([
    { d1: 'a', d2: true, v: 1 },
    { d1: 'a', d2: false, v: 2 },
    { d1: 'b', d2: false, v: 3 },
    { d1: 'b', d2: true, v: 5 }
  ]);

  const wrapper = mount(localVue.extend(ListRow), {
    localVue,
    propsData: {
      id: 'test-list-row',
      dimension: 'd1',
      reduce: 'v'
    },
    attachToDocument: true
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
