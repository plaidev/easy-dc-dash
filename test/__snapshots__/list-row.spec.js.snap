// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list-row to match snapshot 1`] = `
<div class="outer-container" style="width: 377px; height: 233px;">
    <div class="backdrop"></div>
    <div class="card-container">
        <div class="inner-container" style="width: 377px; height: 233px;">
            <div class="container-header">
                <div class="icon-box">
                    <i class="fa fa-window-maximize"></i>
                </div>
                <!---->
            </div>
            <div class="render-area">
                <div id="test-list-row" class="krt-dc-component dc-chart" style="display: flex; position: relative; width: 100%; height: 100%;">
                    <!---->
                    <div class="reset-button">
                        <a class="reset" style="display: none;">
                            <i class="fa fa-ban">&nbsp;</i> <span>フィルター解除</span></a>
                    </div>
                    <!---->
                    <svg width="200" height="200">
                        <g transform="translate(40,20)">
                            <g class="axis" transform="translate(0, 150)">
                                <g class="tick" style="opacity: 0.000001;" transform="translate(0,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">0</text>
                                    <line class="grid-line" x1="0" y1="0" x2="0" y2="-150"></line>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(15,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">1</text>
                                    <line class="grid-line" x1="0" y1="0" x2="0" y2="-150"></line>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(30,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">2</text>
                                    <line class="grid-line" x1="0" y1="0" x2="0" y2="-150"></line>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(45,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">3</text>
                                    <line class="grid-line" x1="0" y1="0" x2="0" y2="-150"></line>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(60,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">4</text>
                                    <line class="grid-line" x1="0" y1="0" x2="0" y2="-150"></line>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(75,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">5</text>
                                    <line class="grid-line" x1="0" y1="0" x2="0" y2="-150"></line>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(90,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">6</text>
                                    <line class="grid-line" x1="0" y1="0" x2="0" y2="-150"></line>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(105,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">7</text>
                                    <line class="grid-line" x1="0" y1="0" x2="0" y2="-150"></line>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(120,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">8</text>
                                    <line class="grid-line" x1="0" y1="0" x2="0" y2="-150"></line>
                                </g>
                                <path class="domain"></path>
                            </g>
                            <g class="row _0" transform="translate(0,5)">
                                <rect width="0" height="67.5" fill="#bd3122" class=""></rect>
                                <text x="10" y="10" dy="0.35em" class="row _0">b</text>
                            </g>
                            <g class="row _1" transform="translate(0,77.5)">
                                <rect width="0" height="67.5" fill="#3182bd" class=""></rect>
                                <text x="10" y="10" dy="0.35em" class="row _1">a</text>
                            </g>
                        </g>
                    </svg>
                </div>
            </div>
        </div>
    </div>
</div>
`;
